app/models/pieces/pawn.rb

```diff
@@ -1,2 +1,27 @@
 class Pawn < Piece
+  def valid_move?(destination_row, destination_column)
+    return false if (destination_column == column_coordinate) && ((destination_row - row_coordinate).abs == 1) &&
+                    square_taken?(destination_row, destination_column)
```

Comment:
You can remove the `square_taken?` check.

We already have it in the `move_to!` method. So we won't need to do that check twice.

I should have written this in the trello card :smile:
The way I'm planning we'll use `valid_move?` is like this

```Ruby
piece.move_to!(row, col) if piece.valid_move?(row_col)
```